flow/connectors/clickhouse/normalize.go

```diff
@@ -390,14 +390,47 @@ func (c *ClickHouseConnector) NormalizeRecords(
 					))
 				}
 			default:
-				projection.WriteString(fmt.Sprintf("JSONExtract(_peerdb_data, '%s', '%s') AS `%s`,", colName, clickHouseType, dstColName))
-				if enablePrimaryUpdate {
-					projectionUpdate.WriteString(fmt.Sprintf(
-						"JSONExtract(_peerdb_match_data, '%s', '%s') AS `%s`,",
-						colName,
-						clickHouseType,
-						dstColName,
-					))
+				projLen := projection.Len()
+				if colType == qvalue.QValueKindBytes {
+					format, err := peerdbenv.PeerDBBinaryFormat(ctx, req.Env)
+					if err != nil {
+						return model.NormalizeResponse{}, err
+					}
+					switch format {
+					case peerdbenv.BinaryFormatRaw:
+						projection.WriteString(fmt.Sprintf("base64Decode(JSONExtract(_peerdb_data, '%s', '%s')) AS `%s`,", colName, clickHouseType, dstColName))
+						if enablePrimaryUpdate {
+							projectionUpdate.WriteString(fmt.Sprintf(
+								"base64Decode(JSONExtract(_peerdb_match_data, '%s', '%s')) AS `%s`,",
+								colName,
+								clickHouseType,
+								dstColName,
+							))
+						}
+					case peerdbenv.BinaryFormatHex:
+						projection.WriteString(fmt.Sprintf("hex(base64Decode(JSONExtract(_peerdb_data, '%s', '%s'))) AS `%s`,", colName, clickHouseType, dstColName))
```

Comment:
might be ideal to send hex to raw table rather in this case, but:
1. that would mean passing down ctx/env somehow to json serialization
2. things would get weird with raw tables being invalidated when changing PEERDB_BINARY_FORMAT
3. base64 is actually a pretty good binary encoding when raw binary isn't available (relatively dense while being computationally straightforward compared to base85, generally available decoding functions on destination)